core/src/main/java/org/jboss/modcluster/load/impl/DynamicLoadBalanceFactorProvider.java

```diff
@@ -84,18 +86,26 @@ public synchronized int getLoadBalanceFactor(Engine engine) {
 
                     totalWeight += weight;
                     totalWeightedLoad += this.average(metricLoadHistory) * weight;
+                } catch (NodeUnavailableException e) {
+                    // The metric requested to put the node into error state
+                    // Call LoadMetric#getLoad on remaining metrics so that historical values are populated
+                    nodeUnavailable = true;
                 } catch (Exception e) {
                     this.log.error(e.getLocalizedMessage(), e);
                 }
             }
         }
 
-        // Convert load ratio to integer percentage
-        int load = (int) Math.round(100 * totalWeightedLoad / totalWeight);
+        if (nodeUnavailable) {
+            return -1;
+        } else {
```

Comment:
I considered this structuring cleaner since following one or other branch clearly shows the type of the outcome. But since you this raises concerns, I removed it.